lib/services/in-memory-db.service.ts

```diff
@@ -67,7 +68,29 @@ export class InMemoryDBService<T extends InMemoryDBEntity> {
   }
 
   /**
-   * Update a record in the `recordMap` of type `T` using the supplied record.
+   * Add the supplied `records` partials array to in-memory data store of records.
+   * Get the `id` of the record by getting the next available `id` value.
+   * Returns a sequential array of the newly generated `ids`.
+   * @param records any array of partial records of type `T` to create
+   */
+  public createMany(records: Array<Partial<T>>): number[] {
+    const newRecords = new Array<number>();
+
+    for (const record of records) {
+      const id = record.id || this.getNextId();
+      const newRecord: T = { ...record, id } as T;
+      this.recordMap = {
+        ...this.recordMap,
+        [id]: newRecord,
+      };
+      newRecords.push(newRecord.id);
+    }
```

Comment:
Would it be a bit cleaner and more functional to use `.map` on the input `records` to return a new array of the created records with the id's? Something like:
```
const created: T[] = records.map((record: Partial<T>) => {
  const id = record.id || this.getNextId();
  const newRecord: T = {
    ...record,
    id
  };
  this.recordMap = {
    ...this.recordMap,
    [id]: newRecord
  };
});
return created;
```

Reply:
Great catch. .map is much cleaner. Will refactor.